public/app/features/trails/DataTrailsApp.tsx

```diff
@@ -99,8 +91,23 @@ let dataTrailsApp: DataTrailsApp;
 
 export function getDataTrailsApp() {
   if (!dataTrailsApp) {
+    const newTrail = newMetricsTrail();
+
+    // Set the initial state of the newTrail based on the URL,
+    // In case we are initializing from an externally created URL or a page reload
+    getUrlSyncManager().initSync(newTrail);
+    // Remove the URL sync for now. It will be restored on the trail if it is activated.
+    getUrlSyncManager().cleanUp(newTrail);
```

Comment:
is this really needed?

Reply:
Yes, in order to find the "recentMatchingTrail" below.

This is part of resolving the "continue same trail on refresh" bug.

Reply:
(unless there is a more straightforward way of populating the trail state from the current url)

Reply:
@darrenjaneczek continue same trail on refresh? I refresh should clear the history

Reply:
It was deemed to be a bug.
With this change, accessing the same URL, either by refresh, or explicit URL, will act similar to selecting the trail from the recent list.